File: design/view_factory/src/main/java/ru/tensor/sbis/design/view_factory/ViewFactory.java
package ru.tensor.sbis.design.view_factory;

import android.view.View;

import androidx.annotation.NonNull;

/**
 * Интерфейс для объявления View Factory
 *
 * @author am.boldinov
 */
public interface ViewFactory<V extends View> {

    /** Метод для создания View элемента. */
    @NonNull V createView();

}

File: design/design_contact_data_view/src/main/java/ru/tensor/sbis/design/contact_data_view/factory/ViewFactory.kt
package ru.tensor.sbis.design.contact_data_view.factory

import android.content.Context
import android.view.ViewGroup
import androidx.annotation.StyleRes
import ru.tensor.sbis.design.contact_data_view.SbisContactPhoneNumberView
import ru.tensor.sbis.design.contact_data_view.R

/**
 * Фабрика по созданию вьюх на [SbisContactDataView]
 */
internal class ViewFactory {

    /** @SelfDocumented **/
    fun createView(context: Context, @StyleRes style: Int): SbisContactPhoneNumberView {
        return SbisContactPhoneNumberView(context, defStyleRes = style).apply {
            id = R.id.design_contact_data_phone_number_view_id
            layoutParams =
                ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        }
    }
}